src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -262,9 +273,32 @@ public void on(final PlayerInteractEvent.EntityInteract event)
     @SubscribeEvent
     public void on(final PlayerInteractEvent.EntityInteractSpecific event)
     {
+        if(isFreeToInteractWith(event.getWorld().getBlockState(event.getPos()).getBlock()))
```

Comment:
Why? Read the EntityInteractSpecific Javadoc:
/**
     * This event is fired on both sides whenever a player right clicks an **entity**.
     *
     * "Interact at" is an interact where the local vector (which part of the entity you clicked) is known.
     * The state of this event affects whether {@link Entity#applyPlayerInteraction} is called.
     * If {@link Entity#applyPlayerInteraction} returns {@link net.minecraft.util.EnumActionResult#SUCCESS}, then processing ends.
     * Otherwise processing will continue to {@link EntityInteract}
     *
     * Canceling the event clientside will cause processing to continue to {@link EntityInteract},
     * while canceling serverside will simply do no further processing.
     */
    @Cancelable
    public static class EntityInteractSpecific extends PlayerInteractEvent

It is for player with entity interaction. Block checking should not be done here!